File: src/main/kotlin/pt/isel/ls/server/data/dataInterfaces/ListData.kt
package pt.isel.ls.server.data.dataInterfaces

import pt.isel.ls.server.utils.BoardList

interface ListData {
    val size: Int

    fun createList(idBoard: Int, name: String): Int

    fun getList(idList: Int, idBoard: Int): BoardList

    fun getListsOfBoard(idBoard: Int, limit: Int, skip: Int): List<BoardList>

    fun deleteList(idList: Int, idBoard: Int)

    fun getListCount(idBoard: Int) : Int
}

File: static-content/data/listData.js
import {fetchReq} from "../utils/utils.js";


async function createList(idBoard, name) {
    return await fetchReq(`board/${idBoard}/list`, "POST", { name: name})
}

async function deleteList(idBoard, idList, action) {
    await fetchReq(`board/${idBoard}/list/${idList}${action != null ? `?action=${action}` : ''}`, "DELETE")
}

export default {
    createList,
    deleteList
}